test/runtests.jl

```diff
@@ -966,6 +967,69 @@ end
 	@test length(jac) == 2
 	@test jac[1] ≈ [ 0.0,  4.0, 12.0]
 	@test jac[2] ≈ [ 1.0,  0.0,  0.0]
+
+    function f_test_1(A, x)
+        u = A*x[2:end] .+ x[1]
+        return u
+    end
+
+    function f_test_2(A, x)
+        u = Vector{Float64}(undef, length(x)-1)
+        u .= A*x[2:end] .+ x[1]
+        return u
+    end
+
+    function f_test_3!(u, A, x)
+        u .= A*x[2:end] .+ x[1]
+    end
+
+    J_r_1(A, x) = Enzyme.jacobian(Reverse, θ -> f_test_1(A, θ), x, Val(5))
+    J_r_2(A, x) = Enzyme.jacobian(Reverse, θ -> f_test_2(A, θ), x, Val(5))
+    J_r_3(u, A, x) = Enzyme.jacobian(Reverse, θ -> f_test_3!(u, A, θ), x, Val(5))
+    
+    J_f_1(A, x) = Enzyme.jacobian(Forward, θ -> f_test_1(A, θ), x)
+    J_f_2(A, x) = Enzyme.jacobian(Forward, θ -> f_test_2(A, θ), x)
+    J_f_3(u, A, x) = Enzyme.jacobian(Forward, θ -> f_test_3!(u, A, θ), x)
+
+    x = ones(6)
+    A = Matrix{Float64}(LinearAlgebra.I, 5, 5)
+    u = Vector{Float64}(undef, 5)
+    @test J_r_1(A, x) == ([1.0, 1.0, 0.0, 0.0, 0.0, 0.0], [1.0, 0.0, 1.0, 0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 1.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0, 1.0, 0.0], [1.0, 0.0, 0.0, 0.0, 0.0, 1.0])
```

Comment:
I wonder if we should change Jacobian to retun the `hcat` of this? Or maybe we can allocate the ouput matrix first and then pass in a view?